### ISSUE TITLE
Indentation regression on multi-line `for` iterator expressions

### ISSUE THREAD

**Issue by mernen** [rust-lang-nursery/rustfmt#998]
rustfmt 0.4 formatted some of my code like the following:

``` rust
        for entry in WalkDir::new(path)
                         .into_iter()
                         .filter_entry(|entry| exclusions.filter_entry(entry)) {
            match entry {
                // ...
```

A bit right-drifted, but still on a perfectly acceptable level to me.

Now, with rustfmt 0.5, this is the result:

``` rust
        for entry in WalkDir::new(path)
            .into_iter()
            .filter_entry(|entry| exclusions.filter_entry(entry)) {
            match entry {
                // ...
```

The indentation change was too aggressive, breaking the visual distinction between the iterator expression and the loop block. Would it be possible to keep at least a double-indentation level for the iterator expression?

**Comment by marcusklaas** [comment]
cc https://github.com/rust-lang-nursery/rustfmt/pull/1050

**Comment by mernen** [comment]
I'm not sure if it was specifically intended or an accidental byproduct of other changes, but 0.8 seems to have fixed this (whereas versions up to 0.7.1 still had the exact same output as the 0.5 regression, despite the referenced issues above). I get precisely what I had proposed:

```rust
        for entry in WalkDir::new(path)
                .max_depth(self.max_depth())
                .follow_links(self.options.follow_links)
                .into_iter()
                .filter_entry(|entry| exclusions.filter_entry(entry)) {
            match entry {
                // ...
```

I'm closing this issue then.
